src/order-truelength.c

```diff
@@ -0,0 +1,227 @@
+#include "order-truelength.h"
+
+// -----------------------------------------------------------------------------
```

Comment:
I think you need a brief explanation here, with links to the equivalent code in base R